anax-in-k8s/cronjobs/auto-upgrade-cronjob.sh

```diff
@@ -245,91 +283,204 @@ function rollback_agent_image() {
     log_debug "rollback_agent_image() end"
 }
 
+# Sets $STATUS_PATH to the path of the next agent upgrade job that failed
+function get_status_path() {
+    log_debug "get_status_path() start"
+
+    # Exit early if /var/horizon/nmp does not exist
+    log_verbose "Checking if /var/horizon/nmp directory exists..."
+    local auto_upgrade_disabled
+    test -d /var/horizon/nmp 2>&1
+    auto_upgrade_disabled=$?
+    if [ "$auto_upgrade_disabled" -eq "1" ]; then
+        log_info "Auto agent upgrade is disabled. /var/horizon/nmp does not exist."
+        write_logs
+        exit 0
+    fi
+
+    # Get the org of the node, exit early if /var/horizon/nmp/{org} does not exist
+    log_verbose "Checking if /var/horizon/nmp/{org} directory exists..."
+    local nmp_org_arr
+    nmp_org_arr=( $(dir /var/horizon/nmp/) )
+    local nmp_org_arr_len
+    nmp_org_arr_len=$(echo ${#nmp_org_arr[@]})
+    if [ "$nmp_org_arr_len" -eq "0" ]; then
+        log_info "There are no active auto upgrade jobs. /var/horizon/nmp has no org directory."
+        write_logs
+        exit 0
+    fi
+    if [ "$nmp_org_arr_len" -ne "1" ]; then
+        write_logs
+        log_fatal 1 "/var/horizon/nmp has multiple org directories: ${nmp_org_arr[*]}"
+    fi
+    local nmp_org
+    nmp_org=${nmp_org_arr[0]}
+
+    # Exit early if no NMP subdirectories exist
+    log_verbose "Searching NMP subdirectories..."
+    local nmp_arr
+    nmp_arr=( $(dir /var/horizon/nmp/$nmp_org/) )
+    local nmp_arr_len
+    nmp_arr_len=$(echo ${#nmp_arr[@]})
+    if [ "$nmp_arr_len" -eq "0" ]; then
+        log_info "There are no active auto upgrade jobs. /var/horizon/nmp/$nmp_org has no NMP directories."
+        write_logs
+        exit 0
+    fi
+
+    # Loop through NMP subdirectories and check each status.json for newest job that needs rollback
+    log_verbose "Getting latest upgrade job status file..."
+    local latest
+    latest=-1
+    local filepath
+    filepath=/var/horizon/nmp/$nmp_org
+    STATUS_PATH=/var/horizon/nmp/fake_org/fake_nmp/status.json
```

Comment:
Are the fake_org and fake_nmp just for testing?

Reply:
I did that just to default it to a file that does not exist, but I can just set it to an empty string. Not sure why I didn't do that before